test/inductor/test_control_flow.py

```diff
@@ -768,6 +772,26 @@ def test_cond_select_with_input_idx(self, device, dynamic):
             dynamic=dynamic,
         )
 
+    @requires_gpu
+    def test_output_on_different_device(self):
+        class FactoryBranches(torch.nn.Module):
```

Comment:
Why not put the model def under CondModels?

Reply:
because I need the .to(GPU_TYPES). I worry this would break if I don't guard it under @requires_gpu